components/org.wso2.carbon.identity.api.server.organization.management/org.wso2.carbon.identity.api.server.organization.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/management/v1/service/OrganizationManagementService.java

```diff
@@ -915,6 +939,59 @@ private OrganizationMetadata getOrganizationMetadataResponse(Organization organi
         return organizationMetadata;
     }
 
+    private MetaAttributesResponse getMetaAttributesResponse(Integer limit, String after, String before, String filter,
+                                                             List<String> metaAttributes, boolean recursive)
+            throws OrganizationManagementServerException {
+
+        MetaAttributesResponse metaAttributesResponse = new MetaAttributesResponse();
+
+        if (limit != 0 && CollectionUtils.isNotEmpty(metaAttributes)) {
+            boolean hasMoreItems = metaAttributes.size() > limit;
+            boolean needsReverse = StringUtils.isNotBlank(before);
+            boolean isFirstPage = (StringUtils.isBlank(before) && StringUtils.isBlank(after)) ||
+                    (StringUtils.isNotBlank(before) && !hasMoreItems);
+            boolean isLastPage = !hasMoreItems && (StringUtils.isNotBlank(after) || StringUtils.isBlank(before));
+
+            String url = "/meta-attributes" + "?limit=" + limit + "&recursive=" + recursive;
+            if (StringUtils.isNotBlank(filter)) {
+                try {
+                    url += "&filter=" + URLEncoder.encode(filter, StandardCharsets.UTF_8.name());
+                } catch (UnsupportedEncodingException e) {
+                    throw new OrganizationManagementServerException(
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getMessage(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getDescription(),
+                            ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL.getCode(), e);
```

Comment:
Need to improve the error message of https://github.com/wso2/identity-organization-management-core/blob/723ee63d56bbda911a26912307b35c8c1c6b667c/components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/constant/OrganizationManagementConstants.java#L424 to use with this endpoint

Reply:
ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL is now updated to 
```
ERROR_CODE_ERROR_BUILDING_PAGINATED_RESPONSE_URL("65020", "Unable to retrieve the paginated results.",
                "Server encountered an error while building paginated response URL."),
```